### DISCUSSION TITLE
Is GetRecords<T>() with row number possible?

### DISCUSSION THREAD

**Discussion by dkini** [JoshClose/CsvHelper#1878] [Q&A]
I have a function to read  CSV files with different class types 
```cs
while (reader.Read())
                {
                    reader.ReadHeader();
                    records = reader.GetRecords<T>()?.ToList();
                }
```
I would like to read the csv row number for each record. Is this possible? I am not sure i can use classmap as the type can be different each time

**Comment by JoshClose** [comment]
Does your class `T` have a property for the index you want to fill? If so, you can map it.
```cs
Map(m => m.RowNumber).Convert(args => args.Row.Parser.Row);
```

If not, you can get it from the reader.
```cs
var rowNumber = reader.Parser.Row;
```
